snapcraft/commands/remote.py

```diff
@@ -193,7 +198,27 @@ def _run_new_or_fallback_remote_build(self, base: str) -> None:
             run_legacy()
             return
 
-        if is_repo(Path().absolute()):
+        current_path = Path().absolute()
+
+        if is_shallow_repo(current_path):
+            emit.debug("Current git repository is shallow cloned.")
+            base = self._get_effective_base()
+            if base in ["core20", "core22"]:
+                emit.progress(
+                    "Remote build for shallow clones is deprecated "
+                    "and will be removed in core24",
+                    permanent=True,
+                )
+                emit.progress("Fallback to legacy remote-build", permanent=True)
+                run_legacy()
+                return
+
+            raise SnapcraftError(
```

Comment:
I'm not sure this exception is possible to reach and it's not clear if it's being tested in the tests.  Furthermore, it's a confusing duplicate of the error on line 256, which also shouldn't be part of the business logic.